### ISSUE TITLE
Redundant Import Warning on Diesel Items

### ISSUE THREAD

**Issue by RyanChristian4427** [rust-lang/rust-clippy#5154]
I'm currently running Clippy against a project I've made using Rocket and Diesel, and I'm getting redundant import warnings, specifically against items from Diesel. 

```
warning: this import is redundant
  --> src/db/auth_repository.rs:10:10
   |
10 | #[derive(Insertable)]
   |          ^^^^^^^^^^
   |
   = note: `#[warn(clippy::single_component_path_imports)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports

warning: this import is redundant
 --> src/models/user.rs:7:10
  |
7 | #[derive(Queryable, Serialize)]
  |          ^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports

warning: this import is redundant
 --> src/schema.rs:1:1
  |
1 | / table! {
2 | |     users (id) {
3 | |         id -> Int4,
4 | |         first_name -> Text,
... |
8 | |     }
9 | | }
  | |_^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports
  = note: this warning originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

I could somehow be misunderstanding, but I'm not sure how a derived attribute could be a redundant import, nor how the schema could be. I stepped away from working on this project for a few months, so I'm not sure when this started, as I only updated all of my tools today.

As requested, Clippy version: `clippy 0.0.212 (ea85b4c 2020-02-01)`

**Comment by flip1995** [comment]
Needs a macro check. cc @xiongmao86

**Comment by xiongmao86** [comment]
Why is this happening? I am still working on the pr, I don't think it is merged.

**Comment by flip1995** [comment]
It's about this PR: #5058

**Comment by xiongmao86** [comment]
Oh, sorry, I mixed it up. I’ll take a look.

**Comment by xiongmao86** [comment]
I think it is probably needs a `in_external_macro` or `from_expansion` check. I took a look at `diesel_derives/src/insertable.rs`,  but I didn't figure out where the redundant import is, and had no idea how to setup a suitable trait and derive code for test.

**Comment by flip1995** [comment]
I'd allow this also in internal macros. So just `utils::in_macro(item.span)` at the beginning of the `check_item` function.

**Comment by xiongmao86** [comment]
Thank you for the tip, @flip1995, do we have to add a test in this case?

**Comment by xiongmao86** [comment]
Thank you for the tip, @flip1995, do we have to add a test in this case?

**Comment by flip1995** [comment]
Yeah, should be a pretty small test case though. just a macro definition and calling this macro:

```rust
macro_rules! m {
    () => {
        use ident;
    };
}

m!()
```

**Comment by xiongmao86** [comment]
My laptop is broken. It would take me at least a couple of days to setup an new enviroment. So I won't reply immediately. If anyone is interesting in this, please take it.

**Comment by JohnTitor** [comment]
@xiongmao86 Sorry for the steal! I'd like to include a fix for this to next clippy up since it's noisy in some environment.

**Comment by xiongmao86** [comment]
@JohnTitor It is no problem, you are welcome to take it.
